### DISCUSSION TITLE
应用datepicker日期选择器的disabled-date方法时禁用特定日期时存在的问题

### DISCUSSION THREAD

**Discussion by HamsterBloke** [element-plus/element-plus#18836] [General]
问题描述：应用datepicker日期选择器的disabled-date方法时，笔者发现**_日历显示日期的遍历次数为2次_**。笔者在onBeforeMount函数中将本地存储数据存储到reactive对象类型数据中，发现日期已经遍历一次，因此考虑在组件实例创建的时候，日期选择器就将当月显示的日期遍历了一次（**_笔者认为是bug_**），请大佬抽出宝贵时间进行验证，如能积极沟通，感激不尽！
目前的解决方案：在disabled-date方法中先用if判断数据存在，再编写禁用特定日期方法
![屏幕截图 2024-11-08 144311](https://github.com/user-attachments/assets/af58293b-cd40-4b0f-8deb-86c636ebf34c)

**Comment by DDDDD12138** [comment]
你的意思是选中选中了某个日期后，又遍历调用了一遍`disabled-date`方法嘛？
